src/main/webapp/search.js

```diff
@@ -0,0 +1,110 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const URL_STRUCTURE = 'https://www.googleapis.com/youtube/v3/search?part=snippet&type=video&maxResults=5&q=';
+const YOUTUBE_API_KEY = 'AIzaSyDYfjWcy1hEe0V7AyaYzgIQm_rT-9XbiGs';
+
+/**
+ * Retrieve the video result from youtube url.
+ */
+function getVideoResults() {
+  const tabResult = document.getElementById('tab_list');
+  const contentResult = document.getElementById('info_list');
+
+  // Add html for tab bar and content bar
+  tabResult.innerHTML = '';
+  contentResult.innerHTML = '';
+
+  url = getRequestUrl();
+  fetch(url).then((response) => response.json()).then((data) => {
+    videos = data.items;
+    videos.forEach(function(video) {
```

Comment:
Have we confirmed that the JSON list returned from the Youtube API is sorted in order of relevance?

(This is hopefully something that would be mentioned somewhere within their documentation).

If not, then we may need to sort these by relevance.

Reply:
This is confirmed with Youtube Data API: https://developers.google.com/youtube/v3/docs/search/list

Reply:
Updated with comments in the code

Reply:
Sounds good, thanks for confirming.